browser/themes/brave_theme_service.cc

```diff
@@ -0,0 +1,41 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/themes/brave_theme_service.h"
+
+#include "brave/browser/themes/theme_properties.h"
+#include "brave/common/pref_names.h"
+#include "chrome/browser/themes/theme_service_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/pref_registry/pref_registry_syncable.h"
+#include "components/prefs/pref_service.h"
+
+// static
+void BraveThemeService::RegisterProfilePrefs(
+    user_prefs::PrefRegistrySyncable* registry) {
+  registry->RegisterIntegerPref(kBraveThemeType, BRAVE_THEME_TYPE_DEFAULT);
+}
+
+// static
+int BraveThemeService::GetBraveThemeType(Profile* profile) {
```

Comment:
I think GetBraveThemeType and SetBraveThemeType should go into brave_theme_service_browsertest.cc or into another file target that is testonly. We don't want people to accidentally use these outside of tests

Reply:
We will have a setting option to switch dark or light. These apis can be used from settings hander.

Reply:
I think we would handle that with a pref observer. In either case I don't think we should call `NotifyThemeChanged` like that outside of tests. It isn't correct because the theme hasn't actually changed at that point

Reply:
I don't mind GetBraveThemeType, but it should return a BraveThemeType type

Reply:
In fact I think we need a pref observer now for this to work correctly. We're basically counting on something else updating the UI and triggering a refresh of the theme values

Reply:
the pref observer should call `NotifyThemeChanged`

Reply:
also the prefs UI isn't set up to deal with something like `SetBraveThemeType`. It just changes the pref value and pref observers are supposed to take care of the rest

Reply:
How about making `SetBraveThemeType` just change prefs and `BraveThemeService` act as our prefs observer?

Reply:
I guess that's fine, I just don't see the point of having `SetBraveThemeType` if it's just setting the pref. We'll never use it in the browser because the UI will set the pref directly.

Reply:
Setter is moved to test file and BraveThemeService observers prefs and will trigger theme change notification.
Getter is used by `MaybeGetDefaultColorForBraveUi()`.